### ISSUE TITLE
VS Code lost all command palette history, again

### ISSUE THREAD

**Issue by fireattack** [microsoft/vscode#220106]
Type: <b>Bug</b>

See: #207706.

So, this just happened to me again. The phenomenon is exactly the same as last time.
I noticed I don't have any MRU in command palette. Checked synced data immediately, for all the history entries of "UI State" I can access, there is no trace of `commandPalette.mru.cache` until the brand new one got added a few minutes ago:

![image](https://github.com/microsoft/vscode/assets/4130991/bbdc657f-f4ef-4afd-8b8a-c2e13106f050)

VS Code version: Code 1.91.0 (ea1445cc7016315d0f5728f8e8b12a45dc0a7286, 2024-07-01T18:52:22.949Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-14600K (20 x 3494)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.77GB (14.63GB free)|
|Process Argv|--crash-reporter-id b6fcf05e-a986-4547-b92a-20ce53e79f08|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (73)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-m3u|af4|1.0.0
vscode-nginx-conf|ahm|0.3.5
Bookmarks|ale|13.5.0
rtf|ale|2.8.0
vscode-json|and|1.5.2
vscode-subtitles|ast|0.4.0
git-exclude|bou|0.4.0
gitignore|cod|0.9.0
curlconverter|cur|0.0.3
githistory|don|0.6.20
xml|Dot|2.5.1
vscode-great-icons|emm|2.1.105
vsc-material-theme|Equ|34.4.0
vsc-material-theme-icons|equ|3.6.1
prettier-vscode|esb|10.4.0
copy-relative-path-and-line-numbers|ezf|0.3.1
vscode-open-in-github|fab|2.3.0
copilot|Git|1.211.0
copilot-chat|Git|0.17.0
remotehub|Git|0.62.0
vscode-github-actions|git|0.26.3
vscode-pull-request-github|Git|0.92.0
go|gol|0.41.4
gc-excelviewer|Gra|4.2.59
beautify|Hoo|1.5.0
vscode-edit-csv|jan|0.9.1
json-escaper|jos|1.1.2
vscode-view-in-browser|kop|1.0.5
autoconf|mae|0.2.0
vscode-autohotkey-plus-plus|mar|5.0.5
rainbow-csv|mec|3.12.0
gitignore|mic|1.0.1
ecdc|mit|1.8.0
autopep8|ms-|2024.0.0
debugpy|ms-|2024.6.0
isort|ms-|2023.10.1
python|ms-|2024.8.1
vscode-pylance|ms-|2024.6.1
remote-containers|ms-|0.375.0
remote-ssh|ms-|0.112.0
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.88.2
azure-repos|ms-|0.38.0
cpptools|ms-|1.20.5
hexeditor|ms-|1.10.0
remote-explorer|ms-|0.4.3
remote-repositories|ms-|0.40.0
remote-server|ms-|1.5.2
copy-json-path|nid|0.4.0
es6-css-minify|olb|3.3.3
postman-for-vscode|Pos|1.1.0
text-power-tools|qcz|1.47.1
seti-icons|qin|0.1.3
sqlite-viewer|qwt|0.5.9
vscode-yaml|red|1.15.0
vscode-sort-json|ric|1.20.0
rust-analyzer|rus|0.3.2020
code-spell-checker|str|3.0.1
vscode-open-in-github|sys|1.18.0
even-better-toml|tam|0.19.2
pdf|tom|1.2.2
simple-rst|tro|1.5.4
sort-lines|Tyr|1.11.0
vscode-lldb|vad|1.10.0
intellicode-api-usage-examples|Vis|0.2.8
vscodeintellicode|Vis|1.3.1
vscode-icons|vsc|12.8.0
volar|Vue|2.0.10
vscode-todo-highlight|way|1.0.5
highlight-bad-chars|wen|0.0.6
word-divider|yut|0.2.1
save-as-root|yy0|1.8.0
cuesheet|zgm|0.4.0

(3 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelc:31048052
dsvsc021:30996838
01bff139:31013167
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
legacy_priority:31082724

```

</details>

**Comment by fireattack** [comment]
Paging @bpasero and @sandy081 to see if there is anything new I can do.

**Comment by fireattack** [comment]
Like last time, I managed to recover my MRU from my work computer's `state.vscdb`, before it got overridden by sync:

![image](https://github.com/microsoft/vscode/assets/4130991/5d78f7d9-e430-461b-ba5c-83a143be7b0c)

**Comment by sandy081** [comment]
Can you please let us know 

- how many machines are getting synced 
- In which machine you did not find the history
- Is it possible to know which machine has removed the history?

**Comment by fireattack** [comment]
I have two computers in active use. One desktop at home and one laptop at work. Both Windows 10. 
I leave my home computer on all the time, and often leave VS Code open for remote tunnel.

I also have a vscode.dev instance on sync, but I never really used it.

It is my home computer that lost the history. The exact scenario is as follows:

* I worked on my working laptop on Friday. Everything is fine. I finished work and put the computer to sleep (VS Code closed, IIRC).
*  When I was back home, and started using VS Code there (which, IIRC, I had left it open for the whole day) for a while, I noticed I lost all my history on this computer, again.
* Just like last time, I immediately started looking for synced data, my local `state.vscdb` and `state.vscdb.backup`, but they all lost the MRU already. 
* Like last time, I cannot find the sync history where **the MRU was removed** -- which is kinda expected since it only lists the last 20 history entries and it refreshes UI state multiple times an hour.
* When I went back to work this Monday, I checked the `state.vscdb` on that computer before opening VS Code. It still has the proper 2000+ MRU histories. I opened VS Code, as expected the MRU was overridden by sync to bad data, but since I backed up `state.vscdb` I recovered my 2000+ MRU manually there.
